### ISSUE TITLE
Chef 12: HWRPs that inherit from LWRPBase

### ISSUE THREAD

**Issue by tas50** [acrmp/foodcritic#575]
From the Chef 12 release notes:

"In Chef Client 11 we had the recipe DSL included in the `Provider` base class. This logic belongs to `LWRPBase` class so for clean up we’ve removed it from `Provider` base class in Chef 12. This change created an issue in lvm cookbook since it has a heavy weight provider that uses the directory resource directly. The workaround is to inherit from `LWRPBase` rather than `Provider` class in order to be able to resources directory in custom resources. We will update our documentation to reflect this change."

**Comment by tas50** [comment]
This one is a bit tricky because poise does it and it's valid there

**Comment by coderanger** [comment]
I can block rules as needed :) I already have a bunch turned off just because ripper can't parse correctly

**Comment by coderanger** [comment]
That said, I don't think this is really a linter level thing, if you've moved from custom resources to libraries/, you're advanced hopefully

**Comment by tas50** [comment]
I've noticed that most of the people that did HWRPs did them wrong so linters can basically poke people to convert their HWRPs over to custom resources because that's what they really want.

**Comment by coderanger** [comment]
Yeah, I kind of wish we have Rubocop-style levels since a general "you seem to have a resource/provider under libraries/, are you sure that's what you want?" rule makes more sense to me than specific analysis bits.
